indico/modules/rb_new/client/js/modules/bookRoom/BookingSuggestion.jsx

```diff
@@ -0,0 +1,130 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Grid, Icon, Label, Message, Popup} from 'semantic-ui-react';
+import {PluralTranslate, Translate, Singular, Param, Plural} from 'indico/react/i18n';
+import {Slot, IndicoPropTypes} from 'indico/react/util';
+
+import Room from '../../components/Room';
+
+import './BookingSuggestion.module.scss';
+
+
+function SuggestionOption({onClick, overrides, icon, text, tooltip}) {
+    const message = (
+        <Message styleName="suggestion-text" size="mini"
+                 onClick={onClick.bind(undefined, overrides)}
+                 warning compact>
+            <Message.Header>
+                <Icon name={icon} /> {text}
+            </Message.Header>
+        </Message>
+    );
+    return (
+        <Popup trigger={message}
+               content={tooltip}
+               position="right center" />
+    );
+}
+
+SuggestionOption.propTypes = {
+    onClick: PropTypes.func.isRequired,
+    overrides: PropTypes.object,
+    icon: PropTypes.string.isRequired,
+    text: IndicoPropTypes.i18n.isRequired,
+    tooltip: IndicoPropTypes.i18n.isRequired
+};
+
+SuggestionOption.defaultProps = {
+    overrides: null
+};
+
+/**
+ * Renders a "suggested room" containing one or more alternative options
+ * e.g. "10 minutes later", "Skip one day"...
+ */
+export default class BookingSuggestion extends React.PureComponent {
+    static propTypes = {
+        /** Room information */
+        room: PropTypes.object.isRequired,
+        /** Suggestions as {time, duration, skip} */
+        suggestions: PropTypes.object.isRequired,
+        /** Will be called when a suggestion is chosen (clicked on) */
+        onClick: PropTypes.func.isRequired
+    };
+
+    renderSuggestionText(room, {time, duration, skip}) {
+        const {onClick} = this.props;
+        return (
+            <>
+                {time && (
+                    <SuggestionOption onClick={onClick}
+                                      overrides={{time}}
+                                      icon="clock"
+                                      text={(
+                                          <PluralTranslate count={time}>
+                                              <Singular>
+                                                  One minute <Param name="modifier" value={time < 0 ? 'earlier' : 'later'} />
+                                              </Singular>
+                                              <Plural>
+                                                  <Param name="count" value={Math.abs(time)} /> minutes{' '}
+                                                  <Param name="modifier" value={time < 0 ? 'earlier' : 'later'} />
```

Comment:
that's not how i18n works :P 

probably need to repeat the whole translate block for the two versions :/

Reply:
Out of curiosity, what is the problem here?

Reply:
It wasn't me writing the original, but you are indeed right.
@mic4ael there are two problems:
 * `earlier` and `later` are not translated;
 * Building sentences from pieces is not great for translation, due to languages that work differently from English (e.g. declensions, etc...). We actually mention this [here](https://github.com/indico/indico/wiki/Internationalisation#translating-variable-sentences).

Reply:
Mhm, right! Thanks!